core/lens/lens_database.ml

```diff
@@ -0,0 +1,286 @@
+open Types
+open Utility
+open Lens_operators
+open Lens_utility
+
+module Constant = Lens_constant
+
+type t = Value.database
+
+class dummy_database = object(_self)
```

Comment:
Maybe "mock" is a better name than "dummy" here?

Reply:
This is internal so I'm not sure it really matters and to me dummy seems self explanatory enough.